### PR TITLE
Add nullability annotations

### PR DESCRIPTION
**Pull request by dbart01**
### What this does
Swift like nullability annotations. This adds them, and also organizes various things.

@bgulanowski @davidmuzi

### CODE REVIEW

File: Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.h [hunk 1]
```diff
@@ -25,8 +25,8 @@
 //
 
 #import <CoreData/CoreData.h>
-
-#import <Buy/BUYObject.h>
+#import "BUYObject.h"
```
Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.h]: Why did you change the import style here? This form may require additional header search paths, instead of relying on the framework implicit search path.

**Comment by bgulanowski** [comment]
There are too many different kinds of changes in this PR. We need PRs to stay focussed. Please break it up.

The import directives in headers in frameworks must use the framework style, because they are consumed by clients that must also use that style.

File: Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.h (hunk 1 above)
Comment (dbart01) [inline Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.h]: For consistency. We always use local-file `#import` style aside from several cases (such as this one). Using the framework-style `#import` also requires that the header is `public` and exposed in the umbrella header, which we may or may not want to do for some files. Using the quotation mark import requires no changes if we decided to go either way.

**Comment by dbart01** [comment]
Correct me if I'm wrong but all this does is reorder some Xcode groups and adds nullability annotations.

**Comment by bgulanowski** [comment]
This PR does three things:

1. Changing import directives (which I'm still against)
2. moving files around in the project organizer
3. modifies interfaces

Would really prefer separate PRs.

**Comment by bgulanowski** [comment]
Headers that are exposed to clients of a framework must use framework import style. We should not require framework clients to add search paths.

**Comment by dbart01** [comment]
Closing in favour of #224 & #225
